### PR TITLE
wxGUI: Allow delete on multiple mapsets in data catalog

### PR DESCRIPTION
**Pull request by lindakladivova**
Added the function delete_mapsets_interactively which is called after selecting more mapsets. Checked possible cases of current or pernament mapset selection.

### CODE REVIEW

File: gui/wxpython/startup/guiutils.py [hunk 1]
```diff
@@ -320,6 +321,170 @@ def delete_mapset_interactively(guiparent, grassdb, location, mapset):
     dlg.Destroy()
 
 
+def delete_mapsets_interactively(guiparent, grassdbs, locations, mapsets):
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:324]: this shouldn't take as input nodes, it should be something like a list of tuples [(db1, loc1, mapset1), (), ...]

File: gui/wxpython/startup/guiutils.py [hunk 2]
```diff
@@ -320,6 +321,170 @@ def delete_mapset_interactively(guiparent, grassdb, location, mapset):
...
+        if mapsets[i].label == "PERMANENT":
+            selected_pernament.append(i)
+            # Create a string for dialog window
+            if len(selected_pernament) < 10:
+                mapset_path = os.path.join(grassdbs[i].label,
+                                            locations[i].label,
+                                            mapsets[i].label)
+                selected_pernament_str = selected_pernament_str + mapset_path + ", "
+            if len(selected_pernament) == 10:
+                selected_pernament_str = selected_pernament_str + "....."
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:351]: this is done with string.join function

File: gui/wxpython/startup/guiutils.py [hunk 3]
```diff
@@ -320,6 +321,170 @@ def delete_mapset_interactively(guiparent, grassdb, location, mapset):
...
+                                            mapsets[i].label)
+                selected_successfully_str = selected_successfully_str + mapset_path + ", "
+            if len(selected_successfully) == 10:
+                selected_successfully_str = selected_successfully_str + "....."
+
+    # Removed last 2 letters in a string
+    if selected_pernament_str:
+        selected_pernament_str = selected_pernament_str[:-2]
+    if selected_successfully_str:
+        selected_successfully_str = selected_successfully_str[:-2]
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:378]: you don't need this when using join

File: gui/wxpython/datacatalog/tree.py [hunk 4]
```diff
@@ -939,17 +942,28 @@ def OnDeleteMap(self, event):
...
-                                        self.selected_grassdb[0].label,
-                                        self.selected_location[0].label,
-                                        self.selected_mapset[0].label)
-        except Exception as e:
-            GError(parent=self,
-                   message=_("Unable to delete mapset: %s") % e,
-                   showTraceback=False)
+        if len(self.selected_mapset) == 1:
+            try:
+                delete_mapset_interactively(self,
```
Comment (petrasovaa) [inline gui/wxpython/datacatalog/tree.py:949]: use data["name"] instead of label everywhere

**Comment by lindakladivova** [comment]
Now, it has only two possible results:

Current or pernament mapset means nothing is deleted:
Screenshot from 2020-07-22 09-56-28

Success:
Screenshot from 2020-07-22 09-54-33

File: gui/wxpython/datacatalog/tree.py [hunk 5]
```diff
@@ -942,13 +944,18 @@ def OnDeleteMap(self, event):
...
         """
         try:
-            delete_mapset_interactively(self,
-                                        self.selected_grassdb[0].data['name'],
-                                        self.selected_location[0].data['name'],
-                                        self.selected_mapset[0].data['name'])
+            db_loc_mapset = []
+            for i in range(len(self.selected_mapset)):
+                # Append to the list of tuples
+                db_loc_mapset.append(tuple(
```
Comment (petrasovaa) [inline gui/wxpython/datacatalog/tree.py:953]: simplify the syntax to `append((db, loc, mapset))`, right now you are creating a list and converting it to tuple.

File: gui/wxpython/startup/guiutils.py [hunk 6]
```diff
@@ -279,44 +280,66 @@ def rename_location_interactively(guiparent, grassdb, location):
...
+
+    # Initialization
+    issues = []
+
+    # Check selected mapsets and remember issues
+    for grassdb, location, mapset in db_loc_mapset:
+
+        # Check for permanent mapsets
+        if mapset == "PERMANENT":
+            issues.append(os.path.join(grassdb, location, mapset) +
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:298]: Use format function as we discussed everywhere.

File: gui/wxpython/startup/guiutils.py [hunk 7]
```diff
@@ -279,44 +280,66 @@ def rename_location_interactively(guiparent, grassdb, location):
...
+                          " is required for a valid location.")
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:299]: Also use translatable strings _()

File: gui/wxpython/startup/guiutils.py [hunk 8]
```diff
@@ -279,44 +280,66 @@ def rename_location_interactively(guiparent, grassdb, location):
...
+    # If any issues, display the warning message and do not delete anything
+    if issues:
+        reasons = "\n".join(issues)
+        dlg = wx.MessageDialog(
             parent=guiparent,
             message=_(
-                "Mapset <PERMANENT> is required for valid GRASS location.\n\n"
-                "This mapset cannot be deleted."
-            ),
+                "Cannot delete one or more mapsets for the following reasons:\n\n"
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:313]: At the end there should be probably a sentence that the deleting of other mapsets wont proceed. Something like "No mapsets will be deleted."

File: gui/wxpython/startup/guiutils.py [hunk 9]
```diff
@@ -279,44 +280,66 @@ def rename_location_interactively(guiparent, grassdb, location):
...
-            "from location <%(location)s>?\n\n"
-            "ALL MAPS included in this mapset will be "
-            "PERMANENTLY DELETED!"
+        dlg.ShowModal()
+    else:
+        dlg = wx.MessageDialog(
+            parent=guiparent,
+            message=_(
+                "Do you want to continue with deleting one or more mapsets?\n\n"
+                "ALL MAPS included in these mapsets will be "
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:326]: This should lit the mapsets that will be deleted.

File: gui/wxpython/startup/guiutils.py [hunk 10]
```diff
@@ -279,44 +280,66 @@ def rename_location_interactively(guiparent, grassdb, location):
     return newlocation
 
 
-def delete_mapset_interactively(guiparent, grassdb, location, mapset):
+def delete_mapset_interactively(guiparent, db_loc_mapset):
```
Comment (petrasovaa) [inline gui/wxpython/startup/guiutils.py:283]: This should return True/False, see #789.

File: gui/wxpython/datacatalog/tree.py (hunk 5 above)
Comment (lindakladivova) [reply gui/wxpython/datacatalog/tree.py:953]: Simplified.

File: gui/wxpython/gis_set.py [hunk 11]
```diff
@@ -649,7 +648,9 @@ def OnDeleteMapset(self, event):
         location = self.listOfLocations[self.lblocations.GetSelection()]
         mapset = self.listOfMapsets[self.lbmapsets.GetSelection()]
         try:
-            delete_mapset_interactively(self, self.gisdbase, location, mapset)
+            db_loc_mapset = []
+            db_loc_mapset.append(tuple([self.gisdbase, location, mapset]))
```
[the remark this answers is not in the document]
Comment (lindakladivova) [reply gui/wxpython/gis_set.py:652]: Simplified as well.

File: gui/wxpython/startup/guiutils.py (hunk 8 above)
Comment (lindakladivova) [reply gui/wxpython/startup/guiutils.py:313]: This sentence added.

File: gui/wxpython/startup/guiutils.py (hunk 6 above)
Comment (lindakladivova) [reply gui/wxpython/startup/guiutils.py:298]: Changed in the whole guiutils.py file.

File: gui/wxpython/startup/guiutils.py (hunk 7 above)
Comment (lindakladivova) [reply gui/wxpython/startup/guiutils.py:299]: Changed.

File: gui/wxpython/startup/guiutils.py (hunk 9 above)
Comment (lindakladivova) [reply gui/wxpython/startup/guiutils.py:326]: It lists all mapsets that will be deleted in terms of their paths. 
Screenshot from 2020-07-23 05-01-48

File: gui/wxpython/startup/guiutils.py (hunk 10 above)
Comment (lindakladivova) [reply gui/wxpython/startup/guiutils.py:283]: I rebased and solved conflict with master because of #789 and considered this as well.

Comment (wenzeslaus) [reply gui/wxpython/startup/guiutils.py]: The merge ended up ignoring the return value, fixed.

**Pull request merged**